developers/academy/js/starter_multimodal_data/_snippets/102_collection.mts

```diff
@@ -0,0 +1,210 @@
+  authCredentials: new weaviate.ApiKey(wcdApikey),
+  headers: requestHeaders
+}
+)
+// END CreateMovieCollection
+
+// Actual instantiation
+
+client.collections.delete("Movie")
+
+// CreateMovieCollection
+await client.collections.create({
+  name: "Movie",
+  properties: [
+    { name: "title", dataType: configure.dataType.TEXT },
+    { name: "overview", dataType: configure.dataType.TEXT },
+    { name: "vote_average", dataType: configure.dataType.NUMBER },
+    { name: "genre_ids", dataType: configure.dataType.INT_ARRAY },
+    { name: "release_date", dataType: configure.dataType.DATE },
+    { name: "tmdb_id", dataType: configure.dataType.INT },
+    { name: "poster", dataType: configure.dataType.BLOB }
+  ],
+  // Define the vectorizer module
+  vectorizers: vectorizer.multi2VecCohere({
+    imageFields: [{ name: "poster", weight: 0.9 }],
+    textFields: [{ name: "title", weight: 0.1 }]
+  }),
+  // Define the generative module
+  generative: configure.generative.cohere(),
+  // END generativeDefinition  // CreateMovieCollection
+})
+
+client.close()
+// END CreateMovieCollection
+
+const weaviateURL = process.env.WCD_URL as string
+const weaviateKey = process.env.WCD_API_KEY as string
+const cohereKey = process.env.COHERE_API_KEY as string
+
+client = await weaviate.connectToWeaviateCloud(weaviateURL, {
+  authCredentials: new weaviate.ApiKey(weaviateKey),
+  headers: {
+    'X-Cohere-Api-Key': cohereKey,  // Replace with your inference API key
+  }
+})
+
+// Get current file's directory when using ESM
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const imgDir = join(__dirname, "images");
+
+
+// BatchImportData
+
+// Instantiate your client (not shown). e.g.:
+// client = weaviate.connectToWeaviateCloud(...) or
+// client = weaviate.connectToLocal(...)
+
+// END BatchImportData
+
+// BatchImportData
+const dataUrl = "https://raw.githubusercontent.com/weaviate-tutorials/edu-datasets/main/movies_data_1990_2024.json"
+const textResponse = await fetch(dataUrl)
+const data = await textResponse.json()
+
+// Create directory if it doesn't exist
+await fs.mkdir(imgDir, { recursive: true });
+
+// Download images
+const postersUrl = "https://raw.githubusercontent.com/weaviate-tutorials/edu-datasets/main/movies_data_1990_2024_posters.zip";
+const postersPath = join(imgDir, "movies_data_1990_2024_posters.zip");
+
+const response = await fetch(postersUrl);
+if (!response.ok) {
+  throw new Error(`HTTP error! status: ${response.status}`);
+}
+const arrayBuffer = await response.arrayBuffer();
+const buffer = Buffer.from(arrayBuffer);
+
+// Write the zip file
+await fs.writeFile(postersPath, buffer);
+
+// Unzip the files
+const zip = new AdmZip(postersPath);
+zip.extractAllTo(imgDir, true);
+
+// Get the collection
+const movies = client.collections.get("Movie")
+
+// Set a counter and initialize Weaviate Object
+let itemsToInsert: Object[] = []
+let counter = 0;
+const MAX_ITEMS = 10;
```

Comment:
Is there a reason we are only importing 10 objects?

Reply:
not anymore!